src/LowercaseSentenceTokenizer.java

```diff
@@ -29,8 +30,19 @@ public class LowercaseSentenceTokenizer implements Tokenizer {
    * @return a list of tokens, where each token is a word or a period
    */
   public List<String> tokenize(Scanner scanner) {
-    // TODO: Implement this function to convert the scanner's input to a list of words and periods
-    return null;
+    List<String> tokens = new ArrayList<>();
+    while (scanner.hasNext()){
+      String token = scanner.next();
+      if (token.substring(token.length()-1).equals(".")){
+        tokens.add(token.substring(0, token.length()-1));
+        tokens.add(token.substring(token.length()-1));
+      } else {
+        tokens.add(token.toLowerCase());
+      }
```

Comment:
Nice logic! I think there's a slight bug though: how would your tokenizer handle this case: "a string that ends with a CAPITAL WORD AND PERIOD."? What's the problem, and how could you fix it?

Also, is there a simpler way you can do line 38 given that you know the token ends in a period if you get there?